### PR TITLE
upgrade api-proxy module to tokio1

### PR DESCRIPTION
**Pull request by dmolokanov**
Upgrades api-proxy and dependecies to tokio1 ecosystem

### CODE REVIEW

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs [hunk 1]
```diff
@@ -4,392 +4,518 @@ use std::future::Future;
 
 /// A [`mqtt3::IoSource`] implementation used by the clients.
 pub struct IoSource {
-	iothub_hostname: std::sync::Arc<str>,
-	iothub_host: std::net::SocketAddr,
-	authentication: crate::Authentication,
-	timeout: std::time::Duration,
-	extra: IoSourceExtra,
+    iothub_hostname: std::sync::Arc<str>,
```
Comment (vadim-kovalyov) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs:7]: is it just formatting?

**Review by vadim-kovalyov — APPROVED** [review]
:shipit:

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs [hunk 2]
```diff
@@ -4,392 +4,518 @@ use std::future::Future;
...
+
+        let (inner, pending_read) = match &mut *self {
+            Io::Raw(stream) => return std::pin::Pin::new(stream).poll_read(cx, buf),
+            Io::WebSocket {
+                inner,
+                pending_read,
+            } => (inner, pending_read),
+        };
+
+        if buf.filled().is_empty() {
```
Comment (arsing) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs:374]: No, the intent of this check is to return early if the provided buffer has no remaining space to write to. So this needs to check `if buf.remaining() == 0`

File: edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs [hunk 3]
```diff
@@ -184,22 +184,23 @@ impl hyper::client::connect::Connection for UnixStream {
 }
 
 impl tokio::io::AsyncRead for UnixStream {
-	fn poll_read(self: std::pin::Pin<&mut Self>, cx: &mut std::task::Context<'_>, buf: &mut [u8]) -> std::task::Poll<std::io::Result<usize>> {
+	fn poll_read(self: std::pin::Pin<&mut Self>, cx: &mut std::task::Context<'_>, buf: &mut tokio::io::ReadBuf<'_>) -> std::task::Poll<std::io::Result<()>> {
```
Comment (arsing) [inline edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs:187]: I'm confused.

1. This crate isn't needed. It only existed to [provide Windows support for UDS.](https://github.com/arsing/hyper-uds/blame/master/README.md#L3)

1. This copy of the crate seems to have taken the inefficient Windows impl that was originally in [windows.rs](https://github.com/arsing/hyper-uds/blob/master/src/windows.rs), moved it into `lib.rs`, and is using it even for Linux. Why?!

**Review by arsing — CHANGES_REQUESTED** [review]

File: edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs (hunk 3 above)
Comment (dmolokanov) [reply edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs:187]: we can ask author of this module... @huguesBouvier do you remember why you did this change?

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs (hunk 1 above)
Comment (dmolokanov) [reply edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs:7]: actually both: changes and formatting

Comment (arsing) [reply edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/io.rs:7]: (You can append `?w=1` to the URL to ignore whitespace. ie https://github.com/Azure/iotedge/pull/4879/files?w=1 )

File: edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs (hunk 3 above)
Comment (dmolokanov) [reply edge-modules/api-proxy-module/rust-sdk/hyper-uds/src/lib.rs]: I removed `hyper-uds` altogether in favor of `hyperlocal`

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs [hunk 4]
```diff
@@ -244,14 +221,19 @@ fn make_hyper_uri(
             Ok(url)
         }
 
-        Scheme::Unix => Ok(hyper_uds::make_hyper_uri(base, path)?),
+        Scheme::Unix => {
+            let host = hex::encode(base.as_bytes());
+            let uri = format!("unix://{}:0{}", host, path);
+            let uri = uri.parse()?;
+            Ok(uri)
```
Comment (arsing) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:228]: https://docs.rs/hyperlocal/0.8.0/hyperlocal/struct.Uri.html

**Review by arsing — CHANGES_REQUESTED** [review]

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs (hunk 4 above)
Comment (dmolokanov) [reply edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/iotedge_client.rs:228]: oh, that's nice!

File: edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/Cargo.toml [hunk 5]
```diff
@@ -6,33 +6,34 @@ license = "MIT"
...
 [dependencies]
-base64 = "0.11"
-bytes = "0.5"
+base64 = "0.13"
+bytes = "1.0"
 futures-core = "0.3"
 futures-channel = "0.3"
 futures-util = "0.3"
-hmac = "0.7"
+hex = "0.4"
```
Comment (arsing) [inline edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/Cargo.toml:14]: Don't need this dep any more.

**Review by arsing — DISMISSED** [review]

**Review by arsing — APPROVED** [review]

**Review by vadim-kovalyov — APPROVED** [review]

**Review by huguesBouvier — APPROVED** [review]

**Review by huguesBouvier — COMMENTED** [review]
Tested the changes

**Pull request merged**